src/Transport/Config/ExtensionMethods/AzureServiceBusTransportExtensions.cs

```diff
@@ -104,6 +104,31 @@ public static void UseNamespaceNamesInsteadOfConnectionStrings(this TransportExt
         }
 
         /// <summary>
+        /// Choose the name of the default namespace.
+        /// </summary>
+        public static void DefaultNamespaceName(this TransportExtensions<AzureServiceBusTransport> transportExtensions, string name )
+        {
+            var settings = transportExtensions.GetSettings();
+            settings.Set(WellKnownConfigurationKeys.Topology.Addressing.DefaultNamespaceName, name);
+        }
+
+        /// <summary>
+        /// Adds a namespace for a specific purpose.
+        /// </summary>
+        public static void AddNamespace(this TransportExtensions<AzureServiceBusTransport> transportExtensions, string name, string connectionString, NamespacePurpose purpose = NamespacePurpose.Partitioning)
```

Comment:
I'd remove the `default` value for `purpose` parameter, to make the choice more explicit.
Or better, I'd create two different methods: one to add namespace for `partitioning` and one to add for `destination only`